Source/CarthageKit/Cartfile.swift

```diff
@@ -165,3 +173,43 @@ extension ResolvedCartfile: CustomStringConvertible {
 			.joined(separator: "")
 	}
 }
+
+
+extension String {
+	
+	/// Returns self without any potential trailing Cartfile comment. A Cartfile
+	/// comment starts with the first `commentIndicator` that is not embedded in any quote
+	var strippingTrailingCartfileComment: String {
+		
+		// since Cartfile syntax doesn't support nested quotes, such as `"version-\"alpha\""`
+		// I can just consider any odd-number occurence of a quote as a quote-start, and any
+		// even-numbered occurrence of a quote as quote-end.
+		// The comment indicator (e.g. `#`) is the start of a comment if it's not nested in quotes.
+		// The following code works also for comment indicators that are are more than one character long (e.g. "//")
```

Comment:
Really like the note on future abstraction here :clap:

But, _apologies for putting our buggy tooling on to you_, some off the maintainers run [SwiftLint](https://github.com/realm/SwiftLint/) which (with our configuration) misparses an actual comment out of the second `//` here and emits a linting warning:

```
Whitespace Comment Start Violation: A comment should always start with a whitespace. (whitespace_comment_start)
```

We could do `"//"` → `"/" + "/"` or maybe, just easier to remove the line all together, considering it’s not describing current behavior…

Sorry to bring you really into the weeds on that one 🙃